### ISSUE TITLE
Critical Bug for saveScoreToReportLater

### ISSUE THREAD

**Issue by danoli3** [nihalahmed/GameCenterManager#52]
Scores are never saved.

Critical and very serious bug.

Issue: 
In ```- (void)saveAndReportScore:(int)score leaderboard:(NSString *)identifier sortOrder:(GameCenterSortOrder)order  { ```
  
```
if ([self checkGameCenterAvailability:NO] == YES) {

// ... EDITED OUT

#if TARGET_OS_IPHONE
        GKScore *gkScore = [[GKScore alloc] initWithLeaderboardIdentifier:identifier];
#else
#ifdef __MAC_10_10
		GKScore *gkScore = [[GKScore alloc] initWithLeaderboardIdentifier:identifier];
#else
		GKScore *gkScore = [[GKScore alloc] initWithCategory:identifier];
#endif
#endif
        // <<--------------- SCORE is never set for gkScore
        [self saveScoreToReportLater:gkScore];
    }

```

Fix: Add  ```[gkScore setValue:score];``` above ``` [self saveScoreToReportLater:gkScore];```

**Comment by danoli3** [comment]
This leads to Sigfault in iOS 9 when trying to report ``` [[GameCenterManager sharedManager] reportSavedScoresAndAchievements];```

As GKScore has no value set (Error).

**Comment by danoli3** [comment]
@nihalahmed Are you around mate? Going to PR this fix.

**Comment by Sam-Spencer** [comment]
Fixed in commit b4d5c3c7f63c0500e9d992d8e5047793814a6dc7
